File: model/apispec/ApiSpec.py
class ApiSpec:
    def __init__(self):
        self.apiSpecId = None  # String
        self.name = None  # String
        self.description = None  # String
        self.contextPath = None  # String
        self.backendServiceUrl = None  # String
        self.status = None  # ApiStatus  # This might be an enum
        self.type = None  # ApiType  # This might be an enum
        self.style = None  # ApiStyle  # This might be an enum
        self.metaData = None  # MetaData  # This might be an enum
        self.policyTemplateId = None  # String
        self.creationFlag = None  # CreationFlag  # This might be an enum
        self.addVersionToContextPath = None  # boolean
        self.authType = None  # AuthenticatorType  # This might be an enum
        self.predicates = None  # List<ApiPolicyFilter>
        self.requestPolicies = None  # Set<ApiPolicyFilter>
        self.responsePolicies = None  # Set<ApiPolicyFilter>
        self.oldVersions = None  # List<ApiSpec>
        self.apiSpecRoutes = None  # Set<ApiSpecRouteModel>
        self.revisions = None  # List<RevisionChangeModel>
        self.associatedResources = None  # List<AssociatedResource>
        self.publications = None  # List<RuntimeConfigPublicationModel>
        self.restriction = None  # RestrictionType  # This might be an enum

File: model/apispec/ApiSpec.java
package com.digitinary.dgate.model.apispec;

import com.digitinary.dgate.model.RestrictionType;
import com.digitinary.dgate.model.apispec.policy.ApiPolicyFilter;
import com.digitinary.dgate.model.authenticator.AuthenticatorType;
import com.digitinary.dgate.model.monetization.AssociatedResource;
import com.digitinary.dgate.model.revision.RevisionChangeModel;
import com.digitinary.dgate.model.route.ApiSpecRouteModel;
import com.digitinary.dgate.model.runtimeconfig.RuntimeConfigPublicationModel;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;
import java.util.Set;

/**
 * The main core model.
 * ApiSpec represent the main API specs deployed/expose by the gateway.
 *
 * @author Salah Abu Msameh
 * @since 04/06/2023
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class ApiSpec {

    private String apiSpecId;
    private String name;
    private String description;
    private String contextPath;
    //private String backendServiceUrl; //  moved to the gateway environment
    private ApiStatus status;
    private ApiType type;
    private ApiStyle style;
    private MetaData metaData;
    private String policyTemplateId;
    private CreationFlag creationFlag;
    private boolean addVersionToContextPath;
    private AuthenticatorType authType;

    private List<ApiPolicyFilter> predicates;
    private Set<ApiPolicyFilter> requestPolicies;
    private Set<ApiPolicyFilter> responsePolicies;
    private List<ApiSpec> oldVersions;
    private Set<ApiSpecRouteModel> apiSpecRoutes;
    private List<RevisionChangeModel> revisions;
    private List<AssociatedResource> associatedResources;
    private List<RuntimeConfigPublicationModel> publications;
    private RestrictionType restriction;
}